### PR TITLE
Adds test for upcoming .NET 8 release

### PR DESCRIPTION
**Pull request by ForestEckhardt**
This is in an effort to prepare for the upcoming release of .NET 8. See this issue for more details https://github.com/paketo-buildpacks/dotnet-core/issues/978.

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: integration/default_apps_test.go [hunk 1]
```diff
@@ -102,42 +102,64 @@ func testDefaultApps(t *testing.T, context spec.G, it spec.S) {
 		})
 
 		context("given a source application with .NET 7", func() {
-			it("should build (and rebuild) a working OCI image", func() {
+			it("should build a working OCI image", func() {
 				var err error
 				source, err := occam.Source(filepath.Join("testdata", "source_7"))
 				Expect(err).NotTo(HaveOccurred())
 
-				for i := 0; i < 3; i++ {
```
Comment (sophiewigmore) [inline integration/default_apps_test.go:110]: what's the reason for this change?

Comment (ForestEckhardt) [reply integration/default_apps_test.go:110]: This is testing a cache use specific case. It does not feel like it is something that is necessary to test for every version of .NET

Comment (sophiewigmore) [reply integration/default_apps_test.go:110]: very well

**Review by sophiewigmore — APPROVED** [review]
